org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/breakpoints/StandardJavaBreakpointEditor.java

```diff
@@ -87,18 +99,60 @@ protected Button createCheckButton(Composite parent, String text) {
 	 *            the parent composite
 	 */
 	protected void createTriggerPointButton(Composite parent) {
-		Composite composite = SWTFactory.createComposite(parent, parent.getFont(), 1, 1, 0, 0, 0);
+		Composite composite = SWTFactory.createComposite(parent, parent.getFont(), 2, 1, 0, 0, 0);
 		fTriggerPointButton = createCheckButton(composite, PropertyPageMessages.JavaBreakpointPage_12);
-
+		fTriggerPointButton.setEnabled(true);
 		fTriggerPointButton.setSelection(isTriggerPoint());
 		fTriggerPointButton.addSelectionListener(new SelectionAdapter() {
 			@Override
 			public void widgetSelected(SelectionEvent event) {
+				if (isTriggerPoint()) {
+					if (isTriggerPoint() && fResumeOnHit.isEnabled()) {
+						fResumeOnHit.setSelection(false);
+						fResumeOnHit.setEnabled(false);
+					} else {
+						fResumeOnHit.setEnabled(true);
+					}
+					if (!fSuspendThread.getSelection() && !fSuspendVM.getSelection()) {
+						fSuspendThread.setSelection(true);
+						setConditionTextToSuspend();
+					}
+				} else {
+					if (fResumeOnHit.isEnabled()) {
+						fResumeOnHit.setSelection(false);
+						fResumeOnHit.setEnabled(false);
+						if (!fSuspendThread.getSelection() && !fSuspendVM.getSelection()) {
+							fSuspendThread.setSelection(true);
+							setConditionTextToSuspend();
+						}
+					} else {
+						fResumeOnHit.setEnabled(true);
+						setConditionTextToSuspend();
+
+					}
+
+				}
+
 				setDirty(PROP_TRIGGER_POINT);
 			}
-
 		});
-
+		fResumeOnHit = SWTFactory.createRadioButton(composite, PropertyPageMessages.BreakpointResumeOnHit, 1);
+		fResumeOnHit.setLayoutData(new GridData());
+		fResumeOnHit.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				if (fResumeOnHit.isEnabled() && javaBpConditionEditor != null) {
```

Comment:
`if (javaBpConditionEditor != null)` is common for both cases and should be extracted as first condition to check